ci-operator/step-registry/cucushift/upgrade/eus/unpause-worker-mcp/cucushift-upgrade-eus-unpause-worker-mcp-commands.sh

```diff
@@ -19,32 +19,75 @@ function debug() {
     fi
 }
 
+# After upgrading, the paused mcp should not be upgraded, so its version is not same with normal mcp
+function pre_check_mcp(){
+    echo "Pre check, after upgrading, the paused mcp should not be upgraded..."
+    local paused_worker_version running_worker_version
+    paused_worker_version=$(oc get -l 'node-role.kubernetes.io/'${PAUSED_MCP_NAME}'=' nodes -ojson | jq '.items[0].status.nodeInfo' | jq -r '.containerRuntimeVersion,.kernelVersion,.osImage')
+    echo "Paused worker version: "
+    echo $paused_worker_version
+    running_worker_version=$(oc get -l 'node-role.kubernetes.io/worker=,node-role.kubernetes.io/'${PAUSED_MCP_NAME}'!=' nodes -ojson | jq '.items[0].status.nodeInfo' | jq -r '.containerRuntimeVersion,.kernelVersion,.osImage')
+    echo "Running worker version: "
+    echo $running_worker_version
+
+    if [[ "$paused_worker_version" == "$running_worker_version" ]]; then
+        echo "Paused worker and normal worker have same version, it is not corect."
+        exit 1
+    fi
+}
+
+# After all mcp upgraded, all worker should on same version
+function post_check_mcp(){
```

Comment:
I think we do not need this check, because we have `cucushift-upgrade-healthcheck` to help us check it.

Reply:
I checked `cucushift-upgrade-healthcheck`, it only checked mcp status, but we also need to check related node version, so I think it's better to keep this method.

Reply:
I prefer to rely on mcp to help us, utilize mcp status to ensure all mcp in a good shape. And the following codes are using hard-code label (`node-role.kubernetes.io/'${PAUSED_MCP_NAME}'=` or `node-role.kubernetes.io/worker=`) to filter nodes, but actually we allow users to define any customized label per their test requirement. That would make the codes hard to maintain, we want the step to be flexiable enough.

If in the future, we found mcp status is not reliable or mcp status can not meet our common requirement, we can enhance the codes in the future to meet all kinds of requirement.

Reply:
1. I think mcp status can only indicate that relevant worker nodes are in good status, so you can see before unpause mcp, both paused and normal mcp are in ready state.
2. In short term, I don't think `node-role.kubernetes.io/worker=` will bring issues.

Reply:
For 1, if you think it is necessary, you can keep it as your desire.

For 2, actually if you looked at https://issues.redhat.com/browse/OCPBUGS-36688, the bug is the source requirements for canary upgrade ci coverage. In the customer case, there is even no any nodes in default worker mcp. That is why I suggest to make the step be flexible enough to avoid any hardcode here, so that we can make the step meet all kinds of cases.

Reply:
ok, let me replace `node-role.kubernetes.io/worker=,` to `node-role.kubernetes.io/master!=,`, I think this will cover the condition in the bug.

btw `node-role.kubernetes.io/'${PAUSED_MCP_NAME}'=` just means a label which prefix is `node-role.kubernetes.io/`, I agree it is not flexible enough, I am not sure if we can just set the label as ${PAUSED_MCP_NAME}. I don't have a cluster in hand, can you help confirm it? cc @jiajliu @shellyyang1989

Reply:
btw, in step `mco-conf-day2-create-custom-mcp`, it also add label with the prefix:
```
run_command "oc label node ${target_nodes_array[*]} node-role.kubernetes.io/$mcp_name="
```

Reply:
ok, let me keep the label `node-role.kubernetes.io/'${PAUSED_MCP_NAME}'=` in all canary upgrade steps.

Reply:
> let me replace node-role.kubernetes.io/worker=, to node-role.kubernetes.io/master!=,, I think this will cover the condition in the bug.

Hold on, let me check it a bit more.

Reply:
> in step mco-conf-day2-create-custom-mcp, it also add label with the prefix:

Good point, if that, let us keep the prefix in this step, in the future, if we need to update them, we can update them together.

Reply:
> Hold on, let me check it a bit more.

I just did some more check in https://issues.redhat.com/browse/OCPBUGS-36688, sounds like the nodes in the new mcp still have `node-role.kubernetes.io/worker=` label, so I guess the code should work well as usual. My fault.

Reply:
`node-role.kubernetes.io/master!=` can find all non-master node, I used this filter in the beginning of the pr. `node-role.kubernetes.io/master!=` is better than `node-role.kubernetes.io/worker=`.